soh/soh/SohMenuBar.cpp

```diff
@@ -249,11 +251,30 @@ void DrawSettingsMenu() {
 
         if (ImGui::BeginMenu("Graphics")) {
         #ifndef __APPLE__
-            if (UIWidgets::EnhancementSliderFloat("Internal Resolution: %d %%", "##IMul", "gInternalResolution", 0.5f, 2.0f, "", 1.0f, true)) {
+            // Disabled on Apple due to a DPI scaling bug.
+            if (UIWidgets::EnhancementSliderFloat("Internal Resolution: %d %%", "##IMul", "gInternalResolution", 0.5f,
+                                                  2.0f, "", 1.0f, true, true,
+                                                  CVarGetInteger("gAdvancedResolution.VerticalResolutionToggle", 0) &&
+                                                      CVarGetInteger("gAdvancedResolution.Enabled", 0) )) {
                 LUS::Context::GetInstance()->GetWindow()->SetResolutionMultiplier(CVarGetFloat("gInternalResolution", 1));
-            };
+            }
             UIWidgets::Tooltip("Multiplies your output resolution by the value inputted, as a more intensive but effective form of anti-aliasing");
         #endif
```

Comment:
There was also a discussion on Discord about the Internal Resolution option being disabled on Apple systems due to a bug with high-DPI support that crops up on systems equipped with Retina displays. This affects the Internal Resolution option, but also N64 Mode and my proposed Advanced Resolution Mode.

I originally had the button to open the window not visible on macOS, but made it visible to allow play-testing as part of the discussion. Let me know if I should hide it again.